packages/auth/src/tenant/routes.ts

```diff
@@ -0,0 +1,138 @@
+import { ParsedUrlQuery } from 'querystring'
+import { AppContext } from '../app'
+import { TenantService } from './service'
+import { BaseService } from '../shared/baseService'
+import { Tenant } from './model'
+
+type TenantRequest<BodyT = never, QueryT = ParsedUrlQuery> = Exclude<
+  AppContext['request'],
+  'body'
+> & {
+  body: BodyT
+  query: ParsedUrlQuery & QueryT
+}
+
+type TenantContext<BodyT = never, QueryT = ParsedUrlQuery> = Exclude<
+  AppContext,
+  'request'
+> & {
+  request: TenantRequest<BodyT, QueryT>
+}
+
+interface CreateTenantBody {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+type UpdateTenantBody = Partial<Omit<CreateTenantBody, 'id'>>
+
+interface TenantParams {
+  id: string
+}
+
+interface TenantResponse {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+export type GetContext = TenantContext<never, TenantParams>
+export type CreateContext = TenantContext<CreateTenantBody>
+export type UpdateContext = TenantContext<UpdateTenantBody, TenantParams>
+export type DeleteContext = TenantContext<never, TenantParams>
+
+export interface TenantRoutes {
+  get(ctx: GetContext): Promise<void>
+  create(ctx: CreateContext): Promise<void>
+  update(ctx: UpdateContext): Promise<void>
+  delete(ctx: DeleteContext): Promise<void>
+}
+
+interface ServiceDependencies extends BaseService {
+  tenantService: TenantService
+}
+
+export function createTenantRoutes({
+  tenantService,
+  logger
+}: ServiceDependencies): TenantRoutes {
+  const log = logger.child({
+    service: 'TenantRoutes'
+  })
+
+  const deps = { tenantService, logger: log }
+
+  return {
+    get: (ctx: GetContext) => getTenant(deps, ctx),
+    create: (ctx: CreateContext) => createTenant(deps, ctx),
+    update: (ctx: UpdateContext) => updateTenant(deps, ctx),
+    delete: (ctx: DeleteContext) => deleteTenant(deps, ctx)
+  }
+}
+
+async function createTenant(
+  deps: ServiceDependencies,
+  ctx: CreateContext
+): Promise<void> {
+  const { body } = ctx.request
+
+  await deps.tenantService.create(body)
+
+  ctx.status = 204
+}
```

Comment:
@mkurapov influenced by the pattern we've been talking about w/ performance. Not 201 with the resource returned. Originally I went with 201 but that really should have a body: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

thus the [204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) for this and the update and delete.